### ISSUE TITLE
DocBlock description is cached incorrectly for sub-classes

### ISSUE THREAD

**Issue by aik099** [FriendsOfPHP/Sami#167]
When PHP file is analyzed for first time (we don't have lexer cache for it) we parse all the structures from it including DocBlocks. When next time cache is used we use just cache.

Consider following code (each class in separate file, but I'm not sure if that matters):

```php
class ClassA {

    /**
     * Comment A.
     */
    public function one() {
    }
}

class ClassB extends ClassA {

    /**
     * Comment B.
     */
    public function one() {
    }
}
```
Then while don't have cache both DocBlock comments are recognized correctly. Next time, when cache is used the `ClassB::one` method would have `Comment A` comment (from parent class) instead of it's own.

I'm not sure if `@param` and `@return` annotations have the same problem or it's just a description part of DocBlock, that is affected.

**Comment by fabpot** [comment]
fixed in f7a4a39

**Comment by aik099** [comment]
Thanks.
